tests/test_it_deploy.py

```diff
@@ -12,13 +12,13 @@ def clear_registry():
 
```

Comment:
Partially unrelated to this PR. I'm finding it a bit difficult to understand the current test organization. The role `__it__` tests, the apparent absence of mapping between code modules and tests...

I would appreciate it if you can explain it a bit and if you envision to reorganize them. (Perhaps in the future, we can add the explanation to a CONTRIBUTING.md file)

Reply:
Maybe let's create an issue for that. Thanks!